screen-wake-lock/wakelock-onrelease.https.html

```diff
@@ -5,17 +5,20 @@
 <script src="/resources/testdriver.js"></script>
 <script src="/resources/testdriver-vendor.js"></script>
 <script>
-async_test(async t => {
+promise_test(async t => {
   await test_driver.set_permission({name: 'screen-wake-lock'}, 'granted', false);
 
   const lock = await navigator.wakeLock.request("screen");
-  lock.onrelease = t.step_func_done((ev) => {
-    assert_class_string(ev, "Event", "release() must fire an Event object");
-    assert_equals(ev.target, lock, "The event's target must be the lock that was acquired");
-    assert_true(ev.isTrusted);
-    assert_false(ev.bubbles);
-    assert_false(ev.cancelable);
+  return new Promise(resolve => {
+    lock.onrelease = t.step_func(ev => {
+      assert_class_string(ev, "Event", "release() must fire an Event object");
+      assert_equals(ev.target, lock, "The event's target must be the lock that was acquired");
+      assert_true(ev.isTrusted);
+      assert_false(ev.bubbles);
+      assert_false(ev.cancelable);
+      resolve();
+    });
```

Comment:
Agree with what @foolip said... can we just go with: 

```suggestion
    lock.onrelease = ev => {
      assert_class_string(ev, "Event", "release() must fire an Event object");
      assert_equals(ev.target, lock, "The event's target must be the lock that was acquired");
      assert_true(ev.isTrusted);
      assert_false(ev.bubbles);
      assert_false(ev.cancelable);
      resolve();
    };
```

or even:

```
    const ev = await new Promise(release => {
      lock.onrelease = release;
    }
    assert_class_string(ev, "Event", "release() must fire an Event object");
    assert_equals(ev.target, lock, "The event's target must be the lock that was acquired");
    assert_true(ev.isTrusted);
    assert_false(ev.bubbles);
    assert_false(ev.cancelable);
 ```
    
```

Reply:
Maybe I'm missing something, but the first suggestion looks like Stephen's original code (which would time out in case any of the assertions fails due to the lack of `t.step_func`), while the second would get stuck on `await` because `lock.release()` is only called later.

Reply:
FWIW I still think the current PR code looks fine, but I guess another possibility that does away with `t.step_func` is the suggestion from #25210:
```js
return new Promise(resolve => {
  lock.onrelease = resolve;
  lock.release();
}).then(event => {
  assert_class_string(...);
  // ...
});
```

Reply:
LGTM!